### ISSUE TITLE
Duplicate Cosmic Settings Desktop Entries

### ISSUE THREAD

**Issue by sebastianrasor** [lilyinstarlight/nixos-cosmic#433]
![image](https://github.com/user-attachments/assets/599cb657-d54d-485f-9760-2e7636181e8e)

This issue might belong in upstream, but I wanted to bring it up here first. I'm not entirely sure what is causing this issue. Very low severity, just a little bit confusing why there's two entries.

**Comment by Thatoo** [comment]
I have the same issue for both Cosmic Settings and Nextcloud Client but no other apps.

**Comment by Elias-Graf** [comment]
Isn't that just the open window when you select it in the launcher it focuses it, and the actual desktop file? I can only reproduce this, if I have a COSMIC Settings window open.

**Comment by Luk45135** [comment]
@Elias-Graf No they are talking about the app library (opened by clicking on Applications or pressing SUPER+A). btw. same issue here.

**Comment by Luk45135** [comment]
> I have the same issue for both Cosmic Settings and Nextcloud Client but no other apps.

For me it also happens with mpv Media Player and Manage Printing.

**Comment by Elias-Graf** [comment]
@Luk45135 Ah sorry, you're right of course.

This happens for me with the Mozilla VPN program as well. @Luk45135 do the programs you have mentioned install some sort of service / is registered in the system configuration and home-manager? Because for the Mozilla VPN, it seems like one of the desktop files comes from enabling the service, and one of them from the client, I added in home-manager.

Maybe, in this flake, we somehow add cosmic-settings from nixpkgs, but also add it through compiling some service / cosmic-comp, or other output?

I would be interested in where the two entries come from (as it appears to be displaying correctly in the cosmic-launcher). But sadly, cosmic-applibrary doesn't currently build for me.

Those issues might also be related:
- https://github.com/pop-os/cosmic-applibrary/issues/169
- https://github.com/pop-os/cosmic-applibrary/pull/122

**Comment by Luk45135** [comment]
> do the programs you have mentioned install some sort of service / is registered in the system configuration and home-manager?

Yes, that seemed to be the case i installed mpv in the system configuration and with home-manager options, but i don't know why that happens with Manage Printing i.e. CUPS
